src/OrbitBase/Profiling.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitBase/Profiling.h"
+
+#include <algorithm>
+#include <limits>
+
+namespace orbit_base {
+int64_t EstimateClockResolution() {
+  int64_t minimum_resolution_found = std::numeric_limits<int64_t>::max();
+
+  // We take the minimum resolution found over 1000 tries. This should already yield stable
+  // results for far fewer iterations (less than 10), but we multiply by a factor of
+  // 100x for robustness. This is harmless in terms of runtime and may cover unforeseen
+  // cases that we can't test easily.
+  constexpr int kIterationsForMinimum = 1000;
+  for (int i = 0; i < kIterationsForMinimum; ++i) {
```

Comment:
Perhaps consider using time, as an additional limit for this? Because e.g, if the resolution is 1ms, then this loop will add 1s to launch time of the service.

Reply:
Done.